docs/conf.py

```diff
@@ -14,44 +16,76 @@
 import sys
 import os
 
+# Define the branch reference for linkcode_resolve
+DOCS_BUILD_REF = os.environ.get("DOCS_BUILD_REF", "stable")
+"""
+DOCS_BUILD_REF: The Git reference used in the GitHub links generated by linkcode_resolve().
+Accepts most git referancees (commit hash or branch name).
+Defaults to 'stable', but can be overridden by setting the DOCS_BUILD_REF environment variable.
+"""
+
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath('../pak/'))
+sys.path.insert(0, os.path.abspath(""".."""))
+sys.path.insert(1, os.path.abspath("""pak"""))
+sys.path.insert(1, os.path.abspath("""tests"""))
```

Comment:
🤔 This shouldn't overwrite, my understanding is that the result of running this code, in regards to the `sys.path` data, would be `[ os.path.abspath(".."), os.path.abspath("tests"), os.path.abspath("pak"), ...rest of sys.path contents... ]`. If I'm not mistaken this allows the code to be sorted by length and maintain the correct order.

# Formal Proof

## Definitions:
1. Let $ sX $ be an arbitrary ordered set of values.
2. Let $ sX' $ be the ordered set of previous values in $ sX $. If $ sX $ is the empty set, then $ sX' $ is also the empty set.
3. Let $ A, B, C $ be distinct values such that:
   - $ A \neq B $
   - $ B \neq C $
   - $ C \neq A $

## Insertion Steps:
1. **Insert $ A $**: 
   - When $ A $ is inserted at the first position in $ sX $, we define the new ordered set as:
   $$sXa = \{ A \} \cup sX'$$
   - Here, $ sXa' $ represents the previous values in $ sXa $, which is equivalent to $ sX' $. The order is maintained by placing $ A $ at the beginning.

2. **Insert $ B $**:
   - Next, we insert $ B $ at the second position in $ sXa' $. The new ordered set becomes:
   $$sXab = \{ A, B \} \cup sX'$$
   - Here, $ sXab' $ represents the previous values in $ sXab $, which is equivalent to $ sX' $. The order is maintained by placing $ B $ immediately after $ A $.

3. **Insert $ C $**:
   - Finally, we insert $ C $ at the second position in $ sXab' $. Since $ B $ is currently at the second position, inserting $ C $ at this position will push $ B $ to the third position. The new ordered set is:
   $$sXacb = \{ A, C, B \} \cup sX'$$
   - Here, $ sXacb' $ represents the previous values in $ sXacb $, which is equivalent to $ sX' $. The order is maintained by placing $ C $ at the second position, resulting in $ B $ being moved to the third position.

## Conclusion:
- The final ordered set after the insertions is:
$$sXacb = \{ A, C, B \} \cup sX'$$
- This shows that the sequence of insertions $ \text{insert}(0, A), \text{insert}(1, B), \text{insert}(1, C) $ results in the ordered set $ \{ A, C, B, sX' \} $.